### DISCUSSION TITLE
Missing images and folders in external library

### DISCUSSION THREAD

**Discussion by lumpiluk** [immich-app/immich#13214] [Q&A] [answered]
I have mounted an external library to `/mnt/external` and successfully imported my photos from 2007 up to November 2023. From that time on until now it seems like only videos have been successfully imported, and I can see no photos from 2024.

What I've tried:
- waited for all jobs to finish
- triple-checked all exclusion rules (listed below)
- clicked "Missing" or "Discover" on all jobs several times; "All" on the library job
- clicked "Scan" on the external library
- clicked "Scan all libraries"
- re-started all containers

My folder structure inside `/mnt/external` looks like this:
- YYYY-MM-dd_some-short-description/
  - (lots of raw and sidecar files)
  - converted/
    - (lots of jpegs)
    - selection_0/
      - (symlinks to selected jpegs)
    - selection_1/
      - …

My exclusion patterns:
```
**/@eaDir/**
**/._*
**/*.cr3
**/*.CR3
**/*.cr2
**/*.CR2
**/selection*/**  # otherwise symlinks will show up as duplicates
**/sequence*/**
**/seq*/**
**/*.fit
**/BL_proxy/**
**/others/**	
**/backup_to-be-sorted/**
**/people/**
**/phone/**
**/profiles/**
**/scripts/**
**/time-lapse/**
**/highlights/**
```

What I'm observing:

In the folder view, I can see that all folders up until 2023-10-20 have the `converted/` subfolder, but none of the folders after it. I also noticed at least one top-level folder missing (2023-10-21).

What am I missing?

**Comment by alextran1502** [comment]
Can you provide one example path of the photo that doesn’t get scanned?

**Reply by lumpiluk** [reply]
A top-level folder that doesn't get scanned:
`/mnt/external/2023-11-10_ogoni9`

A photo that doesn't get scanned even though a video in the parent folder does:
`/mnt/external/2024-01-26_zusammen-gegen-rechts_nie-wieder-ist-jetzt/converted/2024-01-26_zusammen-gegen-rechts_nie-wieder-ist-jetzt_FFF-OPT_lukas-stratmann_LS__09805.jpg`

A photo that gets scanned successfully:
`/mnt/external/2023-09-01_molecule-man/converted/2023-09-01_molecule-man_FFF-OPT_lukas-stratmann_LS__4188.jpg`

It really feels like a lexicographical cutoff after which only videos and their parent folders are included…

Also because it seems like I can successfully add new folders with old photos from 10+ years ago, refresh the library, and they get scanned correctly, but the photos I took yesterday never show up.

**Reply by lumpiluk** [reply]
Two other things that might or might not be relevant:

I think the cutoff date might coincide roughly with the oldest smartphone photo that I have backed up via Immich to the upload folder.

For testing purposes I didn't give the external library full access to `/mnt/external` at first. Instead I provided two subfolders from 2024 as import paths, which at the time were successfully scanned. Later I added `/mnt/external` as import path and deleted the import paths for the two subfolders. The photos from those two folders still show up in my library (as they should), but none of the others between November 2023 and now.

**Comment by lumpiluk** [comment, marked as answer]
Solved it!

TL;DR: Make sure that Immich has read access to all subfolders in your external library.

In one of my subfolders I had some files and folders owned by a different group and user. I deleted them, clicked "scan" on the library again, and now all my photos are found! :)
